src/Event.cpp

```diff
@@ -49,25 +49,25 @@ std::pair<int, bool> Event::get_value(const std::string& name, bool crash_on_fai
     const auto val = event_values_.find(name);
 
     if (val == event_values_.end()) {
-        std::cout << "Value " << name << " not found in event " << static_cast<int>(event_type) << "!" << std::endl;
 
         if (crash_on_fail == true) {
-            std::cerr << "Error: Value not found and program will crash on fail.";
-            exit(EXIT_FAILURE);
+            assert(crash_on_fail);
+            std::cout << "Error: Value not found and program will crash on fail.";
         } else {
+            std::cout << "Value " << name << " not found in event " << static_cast<int>(event_type) << "!" << std::endl;
             return std::make_pair(-1, false);
         }
     }
 
     if (val->second.type_name != typeid(int).name()) {
-        std::cout
-                << "Wrong type int instead of " << val->second.type_name
-                << " in value " << name << " of event " << static_cast<int>(event_type) << "!" << std::endl;
 
         if (crash_on_fail == true) {
-            std::cerr << "Error: Wrong type int and program will crash on fail.";
-            exit(EXIT_FAILURE);
+            assert(crash_on_fail);
```

Comment:
These assert expressions will never be false so assert will never fail. You probably want something like:
```
if(crash_on_fail) {
  assert(val != event_values_.end()); // Assert that Event has key-value pair
  ...
  assert(val->second.type_name == typeid(int).name()); // Assert that value is the correct type
}
```

Note: this is just an example